protocol/amqp/v2/sender.go

```diff
@@ -33,15 +34,13 @@ func (s *sender) Send(ctx context.Context, in binding.Message, transformers ...b
 		return err
 	}
 
-	err = s.amqp.Send(ctx, &amqpMessage)
+	err = s.amqp.Send(ctx, &amqpMessage, s.options)
 	return err
 }
 
 // NewSender creates a new Sender which wraps an amqp.Sender in a binding.Sender
-func NewSender(amqpSender *amqp.Sender, options ...SenderOptionFunc) protocol.Sender {
-	s := &sender{amqp: amqpSender}
-	for _, o := range options {
-		o(s)
-	}
+func NewSender(amqpSender *amqp.Sender, options *amqp.SendOptions) protocol.Sender {
```

Comment:
nit: inconsistent with other functions due to pointer 
```suggestion
func NewSender(amqpSender *amqp.Sender, options amqp.SendOptions) protocol.Sender {
```

Reply:
the thing here is that internally there is a default behavior assumed for a nil passing. from there I am assuming a pointer here. is it wrong?

Reply:
The `Receiver` does not use this though (uses struct semantics). Is this expected? If passing `nil` is valid we

1) either need to document this or
2) use a functional options pattern where we have `nil` as default behavior